core/src/storage/cache.py

```diff
@@ -1,75 +1,41 @@
-from storage.device import get_device_id
-from trezor.crypto import hashlib, hmac, random
+from trezor.crypto import random
 
 if False:
     from typing import Optional
 
 _cached_seed = None  # type: Optional[bytes]
-_cached_seed_without_passphrase = None  # type: Optional[bytes]
-_cached_passphrase = None  # type: Optional[str]
-_cached_passphrase_fprint = b"\x00\x00\x00\x00"  # type: bytes
+_cached_seed_without_passphrase = None  # type: Optional[bytes]  # Needed for SLIP-21
+_cached_session_id = None  # type: Optional[bytes]
```

Comment:
would it make sense to convert some of these to elements in a dict? and have `storage.cache.get`, `set`, `delete`, sort of like `storage.common` does? this would get thrown away when session id changes.

perhaps similarly to the general storage, this should be structured based on app+key?

`session_id` should probably stay separate in any case

Reply:
Check out https://github.com/trezor/trezor-firmware/pull/803/commits/d112c2d266d9d0ffbb6155bd6e7610c199be16d4